code/modules/ai/ai_behaviors/ai_behavior.dm

```diff
@@ -219,6 +236,96 @@ Registers signals, handles the pathfinding element addition/removal alongside ma
 ///Signal handler when the ai is blocked by an obstacle
 /datum/ai_behavior/proc/deal_with_obstacle(datum/source, direction)
 	SIGNAL_HANDLER
+	var/turf/obstacle_turf = get_step(mob_parent, direction)
+
+	for(var/mob/mob_blocker in obstacle_turf.contents)
+		if(!mob_blocker.density)
+			continue
+		//todo: passflag stuff etc
```

Comment:
hi?

Reply:
It's an actual todo - I just moved this block of code from the human level to base ai level